src/JuliaFormatter.jl

```diff
@@ -8,6 +8,9 @@ export format, format_text, format_file, format_dir
 is_str_or_cmd(x::Tokens.Kind) =
     x in (Tokens.CMD, Tokens.TRIPLE_CMD, Tokens.STRING, Tokens.TRIPLE_STRING)
 
+# on Windows lines can end in "\r\n"
+normalize_line_ending(s::AbstractString) = replace(s, "\r\n" => "\n")
```

Comment:
You could also do this here:
```jl
replace(s, r"\n*$", "\n")
```
That should replace zero or more trailing newlines at the end of the file with a single newline. Then you can skip the bit at line 277.

Reply:
regex sensei

Reply:
I've spent a _lot_ of time writing regexes 😬